src/api/certificateAuthority.coffee

```diff
@@ -0,0 +1,56 @@
+Keystore = require('../model/keystore').Keystore
+Certificate = require('../model/keystore').Certificate
+Q = require 'q'
+logger = require 'winston'
+utils = require "../utils"
+pem = require "pem"
+
+exports.createCert = ->
+  certParams = this.request.body
+  keystoreDoc = yield Keystore.findOne().exec()
+  options = certParams
+
+  #This flag determines if this is a CA cert or a client cert
+  if !certParams.selfSigned
```

Comment:
Perhaps we should split this out into two different functions. I think we 'll need something like generateServerCert and generateClientCert api calls. This is also because we would want to do different things with the client certs as opposed to the server cert. The client cert we would save to `keystore.ca` whereas the server cert we would save to `keystore.key` and `keystore.cert.data`. We would also need two different button on the console for these two different functions to.